packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/UploadView.tsx

```diff
@@ -68,12 +70,15 @@ export function UploadView({
   const isAddFolderDisabled = isProcessing || isProcessingComplete;
   const isActionExitDisabled = isProcessing;
   const destinationList = (location.key || '/').split('/');
-
-  const message = isProcessingComplete
+  const actionCompleteMessage = isProcessingComplete
     ? getActionCompleteMessage({
         counts: statusCounts,
       })
     : undefined;
+  const filesValidationMessage =
+    invalidFiles && !isProcessing
+      ? getFilesValidationMessage({ invalidFiles })
```

Comment:
i'm not certain about this but is it better to just have a single `getMessage` function that takes in all the variables and returns a single message as a pattern. I think I remember @calebpollman mentioning this somewhere. 

something like 

```
const message = getMessage({
      isProcessingComplete,
      invalidFiles,
      isProcessing,
      counts
})
```

Reply:
A single `getMessage` would have to accept a lot view states to the callback. It would be more susceptible to breaking changes. I think it's cleaner to expose less view states to the displayText. 

Later we should refactor the message component itself to support multiple messages instead.

Reply:
Yeah, coupling these would make changes more difficult but the display logic is hard to follow right now, so that's the tradeoff. i think we can handle it in a follow up